erpnext/stock/doctype/shipment/shipment.json

```diff
@@ -0,0 +1,478 @@
+   "label": "Notification Details"
+  },
+  {
+   "default": "0",
+   "fieldname": "pickup_from_send_shipping_notification",
+   "fieldtype": "Check",
+   "label": "Send shipping notification"
+  },
+  {
+   "default": "0",
+   "fieldname": "pickup_from_subscribe_to_status_updates",
+   "fieldtype": "Check",
+   "label": "Subscribe to status updates"
+  },
+  {
+   "fieldname": "shipment_notification_subscriptions",
+   "fieldtype": "Table",
+   "label": "Shipment Notification Subscriptions",
+   "options": "Shipment Notification Subscriptions"
+  },
+  {
+   "fieldname": "column_break_27",
+   "fieldtype": "Column Break"
+  },
+  {
+   "default": "0",
+   "fieldname": "delivery_to_send_shipping_notification",
+   "fieldtype": "Check",
+   "label": "Send shipping notification"
+  },
+  {
+   "default": "0",
+   "fieldname": "delivery_to_subscribe_to_status_updates",
+   "fieldtype": "Check",
+   "label": "Subscribe to status updates"
+  },
+  {
+   "fieldname": "shipment_status_update_subscriptions",
+   "fieldtype": "Table",
+   "label": "Shipment Status Update Subscriptions",
+   "options": "Shipment Status Update Subscriptions"
+  },
+  {
+   "fieldname": "parcels_section",
+   "fieldtype": "Section Break",
+   "label": "Parcels"
+  },
+  {
+   "fieldname": "shipment_parcel",
+   "fieldtype": "Table",
+   "label": "Shipment Parcel",
+   "options": "Shipment Parcel"
+  },
+  {
+   "fieldname": "parcel_template",
+   "fieldtype": "Link",
+   "label": "Parcel Template",
+   "options": "Shipment Parcel Template"
+  },
+  {
+   "fieldname": "add_template",
+   "fieldtype": "Button",
+   "label": "Add Template"
+  },
+  {
+   "fieldname": "column_break_28",
+   "fieldtype": "Column Break"
+  },
+  {
+   "fieldname": "shipment_delivery_notes",
+   "fieldtype": "Table",
+   "label": "Shipment Delivery Notes",
+   "options": "Shipment Delivery Notes"
+  },
+  {
+   "fieldname": "shipment_details_section",
+   "fieldtype": "Section Break",
+   "label": "Shipment details"
+  },
+  {
+   "default": "No",
+   "fieldname": "pallets",
+   "fieldtype": "Select",
+   "label": "Pallets",
+   "options": "No\nYes"
+  },
+  {
+   "fieldname": "value_of_goods",
+   "fieldtype": "Currency",
+   "label": "Value of Goods",
+   "precision": "2",
+   "reqd": 1
+  },
+  {
+   "fieldname": "pickup_date",
+   "fieldtype": "Date",
+   "in_list_view": 1,
+   "label": "Pickup Date",
+   "reqd": 1
+  },
+  {
+   "default": "09:00",
+   "fieldname": "pickup_from",
+   "fieldtype": "Select",
+   "label": "Pickup from",
+   "options": "09:00\n09:30\n10:00\n10:30\n11:00\n11:30\n12:00\n12:30\n13:00\n13:30\n14:00\n14:30\n15:00\n15:30\n16:00\n16:30\n17:00\n17:30\n18:00\n18:30\n19:00"
+  },
+  {
+   "default": "17:00",
+   "fieldname": "pickup_to",
+   "fieldtype": "Select",
+   "label": "Pickup to",
+   "options": "09:00\n09:30\n10:00\n10:30\n11:00\n11:30\n12:00\n12:30\n13:00\n13:30\n14:00\n14:30\n15:00\n15:30\n16:00\n16:30\n17:00\n17:30\n18:00\n18:30\n19:00"
```

Comment:
Hardcoding is the not the best approach here.
The start time could be 8 am as well, but my options start at 9 am.
Change the fieldtype to **Time** and perform computations on that. The user must have flexibility to choose the start and end time